src/main/java/org/example/resolver/ProtoGenerator.java

```diff
@@ -0,0 +1,68 @@
+package org.example.resolver;
+
+import org.example.resolver.fileutilities.FileAnalyzer;
+import org.example.resolver.generator.SchemaGenerator;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class ProtoGenerator {
+    private static final String JAVA_EXT = ".java";
+    private static final String FILE_NOT_GENERATED = "File not generated";
+    public void generateAllFiles(String inputDir , String outputDir, int threadCount) throws IOException, ClassNotFoundException {
+        File currDir = new File(inputDir);
+        File[] files = currDir.listFiles();
```

Comment:
This line seems redundant as the iterateAllFiles checks for directories and calls itself with .listFiles()